graphs/graph.cc

```diff
@@ -39,7 +39,7 @@ void Graph::SetNodeSeq(int32_t node_id, const string& seq) {
 void Graph::AddEdge(int32_t source_node_id, int32_t sink_node_id) {
   AssertNodeExists(source_node_id);
   AssertNodeExists(sink_node_id);
-  if (source_node_id >= sink_node_id) {
```

Comment:
We suppose source & sink are different. no?

Reply:
We used to. But now we allow loops!